unblob/processing.py

```diff
@@ -101,16 +101,21 @@ class ExtractionConfig:
     verbose: int = 1
     progress_reporter: Type[ProgressReporter] = NullProgressReporter
 
-    def get_extract_dir_for(self, path: Path) -> Path:
-        """Return extraction dir under root with the name of path."""
+    def intern(self, path: Path) -> Path:
+        """Return path under extract root."""
         try:
             relative_path = path.relative_to(self.extract_root)
         except ValueError:
             # path is not inside root, i.e. it is an input file
             relative_path = Path(path.name)
-        extract_name = path.name + self.extract_suffix
-        extract_dir = self.extract_root / relative_path.with_name(extract_name)
-        return extract_dir.expanduser().resolve()
+        return (self.extract_root / relative_path).expanduser().resolve()
+
+    def get_extract_dir_for(self, carved_path: Path) -> Path:
+        path = carved_path
+        return self.intern(path.with_name(path.name + self.extract_suffix))
+
+    def get_carve_dir_for(self, path: Path) -> Path:
+        return self.intern(path.with_name(path.name + self.carve_suffix))
```

Comment:
Should be something like:

```py
    def get_extract_dir_for(self, path: Path) -> Path:
        return self.intern(path.with_name(path.name + self.extract_suffix))

    def get_carve_dir_for(self, path: Path) -> Path:
        return self.intern(path.with_name(path.name + self.carve_suffix))
```

Reply:
yeah, it was like that, however I wanted to
- say more about the input of `get_extract_dir_for`
- preserve the visual similarity of the logic by not using different variable name

I was not happy with this, so converted as suggested